healthcheck/core.go

```diff
@@ -581,14 +581,21 @@ func (s *Server) notifier() {
 // error and giving up after MaxFailures.
 func (s *Server) send() error {
 	failures := 0
-	for err := s.sendBatch(s.batch); err != nil; {
-		log.Errorf("Send failed: %v", err)
-		failures++
-		if failures > s.config.MaxFailures {
-			return fmt.Errorf("send: %d errors, giving up", failures)
+	for failures < s.config.MaxFailures {
+		err := s.sendBatch(s.batch)
+		if err == nil {
+			break
 		}
+
+		failures++
+		log.Errorf("Send failed: %v, %d times", err, failures)
 		time.Sleep(s.config.RetryDelay)
 	}
+
+	if failures >= s.config.MaxFailures {
```

Comment:
While this works, I think it just complicates things - I'd prefer to see a unconditional for loop, with this moved inside:
```
for {
  err := s.sendBatch(s.batch)
  if err == nil {
    break
  }
  log.Errorf("Send failed %d times: %v", failures, err)

  failures++
  if failures >= s.config.MaxFailures {
    return fmt.Errorf(...)
  }

  time.Sleep(...)
}
```
This avoids having the condition embedded in the code twice.